include/dr/details/device_subrange.hpp

```diff
@@ -0,0 +1,45 @@
+// SPDX-FileCopyrightText: Intel Corporation
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include <dr/concepts/concepts.hpp>
+#include <dr/details/ranges_shim.hpp>
+#include <iterator>
+
+namespace lib {
+
+template <std::forward_iterator I>
+class device_subrange : public rng::subrange<I, I> {
```

Comment:
I added `device_subrange` as a generic subrange we can use inside `segments_tools` while preserving a remote range's rank.  We should perhaps rename this to `remote_subrange`.

Reply:
Sounds good, I will go change it.